apps/console/src/features/applications/utils/sign-in-method-utils.ts

```diff
@@ -317,6 +317,86 @@ export class SignInMethodUtils {
                 idpList: []
             };
         }
+    }
+
+    public static isFederatedConflictWithSMSOTP(
+        args: FederatedConflictWithSMSOTPArgs
+    ): FederatedConflictWithSMSOTPReturnValue {
+
+        const { federatedAuthenticators, steps, subjectStepId } = args;
+
+        /**
+         * We are solving two problems:
+         *
+         * 1) Find out the subject step all federated IdPs:
+         *    We have an array of `federatedAuthenticators` and
+         *    the configured steps. We are only focused on federated
+         *    authentications configured on subject identifier step.
+         *
+         * 2) Walk forward and check other steps (step 2 and beyond) for SMS OTP:
+         *    The reason to do this check is that, if SMS OTP is configured with 
+         *    a federated authenticator, Asgardeo should receive the user's profile
+         *    (including the mobile number) configured on the federated IdP.
+         */
+
+        try {
+
+            /** Start solving the 1st problem **/
+
+            const allOptions: AuthenticatorInterface[] = flatten(
+                steps
+                    .filter(({ id } : { id: number }) => id === subjectStepId)
+                    .map(({ options } : { options: AuthenticatorInterface[] }) => options)
+            );
+
+            /**
+             * Get the list of idps configured in the subject identifier step.
+             */
+            const idPsInSubjectIdStep: GenericAuthenticatorInterface[] =
+                // Extract all the IdP names.
+                [ ...(new Set((allOptions).map(({ idp } : { idp: string }) => idp))) ]
+                    // Find the authenticator model.
+                    .map((idpName: string) => federatedAuthenticators.find(
+                        ({ name } : { name: string }) => name === idpName))
+                    // Remove all the {@code undefined|null} ones please.
+                    .filter(Boolean);
+
+            /** Start solving the 2nd problem **/
+
+            /**
+             * This means that we have only one step, and implies =\>
+             * no SMS OTP is being configured. This is because the interface only
+             * allows SMS OTP to be added to step 2 or beyond.
+             */
+            if (steps.length < 2) {
+                return {
+                    conflicting: false,
+                    idpList: []
+                };
+            }
+
+            const allOtherOptions: AuthenticatorInterface[] = flatten(
+                steps
+                    .slice(1) // Remove the first element (subject identifier step)
+                    .map(({ options } : { options: AuthenticatorInterface[] }) => options) // Get all the options.
```

Comment:
Any specific reason to define params this way? 

I personally find the following to be more readable. WDYT?

```suggestion
                    .map(( options: AuthenticatorInterface[] ) => options) // Get all the options.
```